src/openforms/registrations/contrib/objects_api/plugin.py

```diff
@@ -229,6 +228,67 @@ def get_custom_templatetags_libraries(self) -> list[str]:
             f"{prefix}.objects_api.json_tags",
         ]
 
+    def update_payment_status(self, submission: "Submission", options: dict) -> None:
+        config = ObjectsAPIConfig.get_solo()
+        assert isinstance(config, ObjectsAPIConfig)
+        config.apply_defaults_to(options)
+
+        if not options["payment_status_update_json"]:
+            logger.warning(
+                "Skipping payment status update because not template was configured."
+            )
+            return
+
+        context = {
+            "variables": get_variables_for_context(submission),
+            "payment": {
+                "completed": submission.payment_user_has_paid,
+                "amount": submission.payments.sum_amount(),
+                "public_order_ids": ",".join(
+                    submission.payments.get_completed_public_order_ids()
+                ),
+            },
+        }
+
+        # FIXME: replace with better suited alternative dealing with JSON specifically
+        updated_record_data = render_from_string(
+            options["payment_status_update_json"],
+            context=context,
+            disable_autoescape=True,
+            backend=openforms_backend,
+        )
+        self._check_size_rendered_json(updated_record_data)
+
+        with catch_json_decode_errors():
+            updated_record_data = json.loads(updated_record_data)
```

Comment:
I think it'd be good to have an anti-corruption layer that will allow us to refactor in the future, so wrapping this (here) and in the `register` hook into something with the following signature would be useful:

```py
def render_to_json(template: str, context: dict[str, Any]) -> JSONValue:
    """Render the provided template and context into a JSON serializable object"""
    ...
```

we can then in the future change this implementation with a better solution than Django templates.